ctc_decoder/beam_search.py

```diff
@@ -89,23 +95,29 @@ def beam_search(mat: np.ndarray, labels: str, beam_width: int = 25, lm: Optional
         for labeling in best_labelings:
 
             # probability of paths ending with a non-blank
-            pr_non_blank = 0
+            pr_non_blank = LOG_ZERO
             # in case of non-empty beam
             if labeling:
                 # probability of paths with repeated last char at the end
-                pr_non_blank = last.entries[labeling].pr_non_blank * mat[t, labeling[-1]]
+                if last.entries[labeling].pr_non_blank == LOG_ZERO:
+                    pr_non_blank = np.log(mat[t, labeling[-1]])  # cannot add to -inf
+                else:
+                    pr_non_blank = last.entries[labeling].pr_non_blank + np.log(mat[t, labeling[-1]])
 
             # probability of paths ending with a blank
-            pr_blank = last.entries[labeling].pr_total * mat[t, blank_idx]
+            if last.entries[labeling].pr_total == LOG_ZERO:
+                pr_blank = np.log(mat[t, blank_idx]) # cannot add to -inf
```

Comment:
directly using np.log creates warning outputs for np.log(0).
Define some wrapper function which takes care of that:
```
def log(x):
    with np.errstate(divide='ignore'):
        return np.log(x)
```